subprojects/core/src/main/java/org/gradle/initialization/BuildOperatingFiringTaskExecutionPreparer.java

```diff
@@ -67,17 +76,52 @@ public void run(BuildOperationContext buildOperationContext) {
                     return toTaskPaths(taskGraph.getFilteredTasks());
                 }
 
-                private List<String> toTaskPaths(Set<Task> tasks) {
-                    return ImmutableSortedSet.copyOf(Collections2.transform(tasks, new Function<Task, String>() {
-                        @Override
-                        public String apply(Task task) {
-                            return task.getPath();
+                @Override
+                public List<PlannedTask> getTaskPlan() {
+                    List<Node> scheduledWork = taskGraph.getScheduledWork();
+                    Set<DefaultPlannedTask> taskPlan = toPlannedTasks(scheduledWork);
+                    return new ArrayList(taskPlan);
+                }
+
+                private Set<DefaultPlannedTask> toPlannedTasks(List<Node> scheduledWork) {
```

Comment:
Why not return `Set<PlannedTask>` from here? I also wonder why you need to use a `Set` here, and not a `List`.

Reply:
fixed